### PR TITLE
Clean up crystal implementation

### PR DESCRIPTION
**Pull request by RX14**
I ran `crystal tool format` on the code, then cleaned up the implementation to be a bit more idiomatic crystal. Running time shouldn't change, this is only cosmetic.

### CODE REVIEW

File: crystal/kmeans.cr [hunk 1]
```diff
@@ -52,15 +47,13 @@ def clusters(xs : Array(Point), centroids : Array(Point))
...
 
-  i = 0
-  while i < (iters - 1)
+  ITERS.times do
     centroids = clusters(xs, centroids).map { |l| average(l) }
-    i += 1
   end
 
-  clusters(xs, centroids).map { |l| average(l) }
+  centroids
```
Comment (RX14) [inline crystal/kmeans.cr:56]: While this is equivalent to the previous implementation, this is not equivalent to the scala implementation as it does not run the final `average`

File: crystal/kmeans.cr [hunk 2]
```diff
@@ -40,46 +35,43 @@ def dist(x : Point, y : Point)
...
 def run(xs : Array(Point))
-  centroids = xs.first(n)
+  centroids = xs.first(N)
 
-  i = 0
-  while i < (iters - 1)
-    centroids = clusters(xs , centroids) .map { | l | average(l) }
-    i += 1
+  ITERS.times do
+    centroids = clusters(xs, centroids).map { |l| average(l) }
```
Comment (Sija) [inline crystal/kmeans.cr:53]: Could be rewritten as `centroids = clusters(xs, centroids).map &->average(Array(Point))`.

Comment (RX14) [reply crystal/kmeans.cr:53]: Which is uglier.

Comment (Sija) [reply crystal/kmeans.cr:53]: But should be faster since it doesn't allocate additional closure.

Comment (RX14) [reply crystal/kmeans.cr:53]: @Sija no, this block isn't captured so there's no performance difference.

**Pull request merged**